rust/kernel/delay.rs

```diff
@@ -56,6 +56,18 @@ pub fn coarse_sleep(duration: Duration) {
     unsafe { bindings::msleep(coarse_sleep_conversion(duration)) }
 }
 
+/// busy wait for enough loop cycles to achieve the desired delay.
+///
+/// This function supports the C side `mdelay`, `udelay`, and `ndelay` functions.
```

Comment:
It doesn't look like it ever actually calls `mdelay`?

Reply:
Oops, `mdelay` is just a wrapper of `udelay`. So I tried to use `udelay`. But I used `subsec_micros` so it doesn't worked.
`mdelay` calls `udelay` with a trick to avoid overflows. Copying the trick in Rust isn't good so I simply use `mdelay` now.